src/com/phpuaca/completion/MockedClassTypeProvider.java

```diff
@@ -0,0 +1,232 @@
+        phpNamedElements.add(phpNamedElement);
+
+        parameter = PhpTypeProviderUtil.getResolvedParameter(phpIndex, parameter);
+        if (parameter == null) {
+            return phpNamedElementCollections;
+        }
+
+        for (PhpMethodSignature matchedSignature : signatures) {
+            if (PhpUtil.isCallTo((Method) phpNamedElement, matchedSignature.getClassName(), matchedSignature.getMethodName())) {
+                Collection<? extends PhpNamedElement> namedElements = getByParameter(project, parameter);
+                phpNamedElements.addAll(namedElements);
+            }
+        }
+
+        return new ArrayList<PhpNamedElement>(phpNamedElements);
+    }
+
+    private List<PhpMethodSignature> getSignatures() {
+        List<PhpMethodSignature> signatures = new ArrayList<PhpMethodSignature>();
+
+        signatures.add(new PhpMethodSignature("\\PHPUnit_Framework_MockObject_MockBuilder", "getMock", 0));
+        signatures.add(new PhpMethodSignature("\\PHPUnit_Framework_TestCase", "getMock", 0));
+        signatures.add(new PhpMethodSignature("\\PHPUnit_Framework_TestCase", "getMockForAbstractClass", 0));
+        signatures.add(new PhpMethodSignature("\\PHPUnit_Framework_TestCase", "getMockForTrait", 0));
+        signatures.add(new PhpMethodSignature("\\PHPUnit_Framework_TestCase", "prophesize", 0));
+        signatures.add(new PhpMethodSignature("\\Prophecy\\Prophet", "prophesize", 0));
+        signatures.add(new PhpMethodSignature("\\Prophecy\\Prophecy\\ObjectProphecy", "reveal", 0));
+
+        return signatures;
+    }
+
+    private ArrayList<PhpMethodSignature> getMatchedSignatures(String methodName, List<PhpMethodSignature> methodSignatureSettingList) {
+
+        ArrayList<PhpMethodSignature> matchedSignatures = new ArrayList<PhpMethodSignature>();
+
+        for(PhpMethodSignature methodSignatureSetting: methodSignatureSettingList) {
+            if(methodSignatureSetting.getMethodName().equals(methodName)) {
+                matchedSignatures.add(methodSignatureSetting);
+            }
+        }
+
+        return matchedSignatures;
+    }
+
+    private String buildParameterMethodSignature(String refSignature, int parameterIndex, PsiElement[] parameters) {
+        if (parameters.length - 1 < parameterIndex) {
+            return null;
+        }
+        PsiElement parameter = parameters[parameterIndex];
+        if ((parameter instanceof StringLiteralExpression)) {
+            String param = ((StringLiteralExpression)parameter).getContents();
+            if (StringUtil.isNotEmpty(param)) {
+                return refSignature + TRIM_KEY + param;
+            }
+        }
+
+        if (parameter instanceof PhpReference && (parameter instanceof ClassConstantReference || parameter instanceof FieldReference)) {
+            String signature = ((PhpReference) parameter).getSignature();
+            if (StringUtil.isNotEmpty(signature)) {
+                if (signature.startsWith("#K#C")) {
```

Comment:
I can’t believe that there is no way to develop this functionality without strange string operations.